storage/innobase/row/row0umod.cc

```diff
@@ -1141,11 +1141,6 @@ row_undo_mod_upd_exist_sec(
 			externally stored column. */
 			ut_a(dict_table_has_atomic_blobs(index->table));
 
-			/* This is only legitimate when
-			rolling back an incomplete transaction
-			after crash recovery. */
-			ut_a(thr_get_trx(thr)->is_recovered);
-
 			/* The server must have crashed before
 			completing the insert of the new
 			clustered index entry and before
```

Comment:
Also the first part of the following comment is needs to be revised:
InnoDB must have run out of space or been killed before …